packages/yarnpkg-pnpify/sources/NodeModulesFS.ts

```diff
@@ -374,12 +384,22 @@ class PortableNodeModulesFs extends FakeFS<PortablePath> {
   async readdirPromise(p: PortablePath, opts: {withFileTypes: boolean}): Promise<Array<Filename> | Array<Dirent>>;
   async readdirPromise(p: PortablePath, {withFileTypes}: {withFileTypes?: boolean} = {}): Promise<Array<string> | Array<Dirent>> {
     const pnpPath = this.resolvePath(p);
-    if (!pnpPath.resolvedPath) {
-      throw PortableNodeModulesFs.createFsError('ENOENT', `no such file or directory, scandir '${p}'`);
-    } else if (pnpPath.dirList) {
-      if (withFileTypes)
-        throw new Error(`Unimplemented option withFileTypes w/ n_m`);
-      return Array.from(pnpPath.dirList);
+    if (pnpPath.dirList || this.resolvePath(ppath.join(p, toFilename('node_modules'))).dirList) {
+      let fsDirList: Filename[] = [];
+      try {
+        fsDirList = await this.baseFs.readdirPromise(pnpPath.resolvedPath);
+      } catch (e) {
+        // Ignore errors
+      }
```

Comment:
Mentioned on Discord: some packages (like `fsevents`) seem to generate their own physical node_modules folders somehow, which then need to be concatenated with the virtual content.